libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs

```diff
@@ -95,6 +95,17 @@ public static async Task RunAsync(this Dialog dialog, ITurnContext turnContext,
             }
         }
 
+        /// <summary>
+        /// Obtain the CultureInfo in DialogContext.
+        /// </summary>
+        /// <param name="dialogContext">The dialogContext to extract information.</param>
+        /// <returns>A string representing the current locale.</returns>
+        public static string GetLocale(this DialogContext dialogContext)
+        {
+            var locale = (dialogContext.Context as TurnContext).Locale ?? null;
+            return locale;
```

Comment:
Avoid soft casts, it will return null if Context is not of the expected type and you will get a null ref exception.

I would refactor this line of code as:
return ((TurnContext)dialogContext.Context).Locale

But see my note about adding a new Locale property in TurnContext

If you decide to keep this method add unit tests for it.

Reply:
I think this is followed by the previous comment Tom gave, and sure @cosmicshuai we should add UT for this.

Reply:
removed the soft cast here.

Reply:
Did you add the tests?

Reply:
I didn't find a appropriate place to add a unit test for this. Do you have any suggestion for this? @gabog 
But I have a integration test for this logic in Adaptive-Templates-Tests.

Reply:
The convention is to put unit tests un a file that has the same name of the class being tested + test. If we don't have, we should create one.
Assuming you moved the code into dialogcontext, the test should be in a file called DialogContextTests.cs

Reply:
Got it. I will create a new PR to add test for this.